starlite/template/base.py

```diff
@@ -21,8 +28,25 @@ def url_for(context: Mapping[str, Any], route_name: str, **path_parameters: Any)
     Returns:
         A fully formatted url path.
     """
-    request = cast("Request", context.get("request"))
-    return request.app.route_reverse(route_name, **path_parameters)
+    return context["request"].app.route_reverse(route_name, **path_parameters)
+
+
+def csrf_token(context: TemplateContext) -> str:
+    """Sets a CSRF token on the template.
+
+    Notes:
+        - to use this function make sure to pass an instance of [CSRFConfig][starlite.config.csrf_config.CSRFConfig] to
+        the [Starlite][starlite.app.Starlite] constructor.
+
+    Args:
+        context: The template context.
+
+
+    Returns:
+        A CSRF token if the app level `csrf_config` is set, otherwise an empty string.
+    """
+    csrf_config = context["request"].app.csrf_config
+    return generate_csrf_token(csrf_config.secret) if csrf_config else ""
```

Comment:
I might be wrong here but it looks like this helper should read token from cookies.

Because otherwise token in cookies and token in templates (hidden field) will be different